p2p/server.go

```diff
@@ -225,7 +225,8 @@ type Server struct {
 	checkpointAddPeer       chan *conn
 
 	// State of run loop and listenLoop.
-	inboundHistory expHeap
+	inboundHistory     expHeap
+	disconnectEnodeSet map[enode.ID]struct{}
```

Comment:
Unless the user spams `admin_removePeer`, then it would use up a lot of memory. My rough estimation is as follows:

Assuming each node removes 3000 peers (this number should be considered an overestimation, because the default MaxPeer in the config file is 200 only):

Memory for keys:
3000 entries * 32 bytes/entry (enode.ID is fixed 32 bytes) 
= 96000 bytes

Memory for map overhead (rough estimate):
3000 entries * (8 bytes/pointer to key + 8 bytes/pointer to value + 8 bytes/map internal data structure)
= 3000 entries * (24 bytes/entry)
= 72000 bytes

Total:
= 96000 bytes + 72000 bytes
= ~164KB

I think the number is still pretty reasonable.